skimage/transform/_thin_plate_splines.py

```diff
@@ -0,0 +1,331 @@
+
+    def _radial_distance(self, points):
+        """Compute the pairwise radial distances of the given points to the control points.
+
+        Parameters
+        ----------
+        points : ndarray
+            N points in the source space
+        Returns
+        -------
+        ndarray :
+            The radial distance for each `N` point to a control point.
+        """
+        dist = sp.spatial.distance.cdist(self.src, points)
+        return _radial_basis_function(dist)
+
+
+def _radial_basis_function(r):
+    """Compute basis kernel function for thine-plate splines.
+
+    Parameters
+    ----------
+    r : ndarray
+        Input array representing the norm distance between interlandmark
+        distances for the source form and based on the (x,y) coordinates
+        for each of these points.
+    Returns
+    -------
+    ndarray
+        Calculated kernel function U.
+    """
+    _small = 1e-8  # Small value to avoid divide-by-zero
+    return np.where(r == 0.0, 0.0, (r**2) * np.log((r**2) + _small))
+
+
+def _ensure_2d(arr):
+    """Ensure that `array` is a 2d array.
+
+    In case given 1d array, expand the last dimension.
+    """
+    array = np.asarray(arr)
+
+    if array.ndim not in (1, 2):
+        raise ValueError("Array must be be 2D.")
+    # Expand last dim in order to interpret this as (n, 1) points
+    if array.ndim == 1:
+        array = array[:, None]
+    if array.size == 0:
+        raise ValueError("Array of points can not be empty.")
+    if len(array) < 3:
+        raise ValueError("Array points less than 3 is undefined.")
+    return array
+
+
+def tps_warp(
+    image,
+    src,
+    dst,
+    output_region=None,
+    interpolation_order=1,
+    grid_scaling=None
+):
+    """Return an array of warped images.
+
+    Define a thin-plate-spline warping transform that warps from the
+    src to the dst, and then warp the given images by
+    that transform.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image.
+    src : (N, 2)
+        Control point at source coordinates.
+    dst : (N, 2)
+        Control point at target coordinates.
+    output_region : tuple of integers, optional
+        The region ``(xmin, ymin, xmax, ymax)`` of the output
+        image that should be produced. (Note: The region is inclusive, i.e.
+        xmin <= x <= xmax)
+    interpolation_order : int, optional
+        If value is 1, then use linear interpolation else use
+        nearest-neighbor interpolation.
+    grid_scaling : int, optional
+        If grid_scaling is greater than 1, say x, then the transform is
+        defined on a grid x times smaller than the output image region.
+        Then the transform is bilinearly interpolated to the larger region.
+        This is fairly accurate for values up to 10 or so.
+
+    Returns
+    -------
+    warped : array_like
+        The warped input image.
+
+    Raises
+    ------
+    RuntineError
+        If ``x_steps`` or ``y_steps`` <= 0
```

Comment:
```suggestion
```
In general, its a good instinct to document the behavior as extensively as possible. Though in this case, I think it is helpful because the variables `x_steps` and `y_steps` aren't actually introduced. See also my other comment.

Also typo in "RuntineError".